addons/mail/static/src/discuss/core/public_web/discuss_sidebar_categories.xml

```diff
@@ -34,6 +34,7 @@
             <DiscussSidebarChannel t-foreach="filteredThreads(category.threads)" t-as="thread" t-key="thread.localId" thread="thread"/>
         </t>
         <DiscussSidebarChannel t-elif="store.discuss.thread?.in(category.threads)" thread="store.discuss.thread"/>
+        <DiscussSidebarChannel t-elif="store.discuss.thread?.parent_channel_id?.in(category.threads)" isCategoryOpen="false" thread="store.discuss.thread.parent_channel_id" />
```

Comment:
`isCategoryOpen="false"` enforces showing of all subchannels when folding the category.
The purpose of folding category is to fold everything but the active item, so it should only show the main channel and the active sub channel.

By removing `isCategoryOpen` and adapting the condition for showing sub channels with `thread.discussAppCategory.open`, this will do the trick